lib/documents.js

```diff
@@ -2838,7 +2844,22 @@ function onReadAllInit(output) {
       break;
   }
   jobState.startTime = Date.now();
-  for (let i = 0; i < maxRequesters; i++) {
+
+  if(jobState.jobOptions.consistentSnapshot){
+    jobState.consistentSnapshotTimestamp = new mlutil.Timestamp();
+    jobState.requesterCount++;
+    let firstRequestCompleted = onReadAllDocs(jobState,0);
+    if(firstRequestCompleted){
+      spinReaderThreads(jobState,1, maxRequesters);
```

Comment:
Hi Erik, this is one of the ways to make the code synchronous. The variable "firstRequestCompleted" will be true only after 2851 is executed, so line 2853 is always executed after the first request. I had put loggers in the function readDocumentsImpl for timestamp values in each request and only the first value came as null. Rest all requests had values in them.

Reply:
Hi, Anushree:  onReadAllDocs() will certainly have executed.

But, executing onReadAllDocs() only starts the first request.  The response from the first request won't come back until later when the object or chunked mode response callback executes.

Because the timestamp is in the response, the timestamp won't be available until the response comes back.

If onReadAllDocs() returns a value, the firstRequestCompleted variable will evaluate to true, but that doesn't guarantee the timestamp is available.

Am I missing something?